MetafileImporter/Logic/vtkSlicerMetafileImporterLogic.cxx

```diff
@@ -489,6 +524,173 @@ vtkMRMLNode* vtkSlicerMetafileImporterLogic::ReadSequenceMetafileImages( const s
+  if ( sliceNode == NULL || sliceNode->GetImageData() == NULL )
+  {
+    return;
+  }
+  vtkImageData* sliceImageData = sliceNode->GetImageData();
+  if ( sliceImageData->GetDimensions()[ 0 ] == 0 || sliceImageData->GetDimensions()[ 1 ] == 0 || sliceImageData->GetDimensions()[ 2 ] != 1 )
+  {
+    return;
+  }
+
+  // Allocate the output image data from the first slice
+  vtkSmartPointer< vtkImageData > imageData = vtkSmartPointer< vtkImageData >::New();
+  int* sliceDimensions = sliceImageData->GetDimensions();
+  double* sliceSpacing = sliceImageData->GetSpacing();
+  imageData->SetDimensions( sliceDimensions[ 0 ], sliceDimensions[ 1 ], imageSequenceNode->GetNumberOfDataNodes() );
+  imageData->SetSpacing( sliceSpacing[ 0 ], sliceSpacing[ 1 ], sliceSpacing[ 2 ] );
+  imageData->SetOrigin( 0, 0, 0 );
+
+#if (VTK_MAJOR_VERSION <= 5)
+  imageData->SetScalarType(sliceImageData->GetScalarType());
+  imageData->SetNumberOfScalarComponents(sliceImageData->GetNumberOfScalarComponents());
+  imageData->AllocateScalars();
+#else
+  imageData->AllocateScalars(sliceImageData->GetScalarType(), sliceImageData->GetNumberOfScalarComponents());
+#endif
+
+  int sliceSize=sliceImageData->GetIncrements()[2];
+  for ( int frameNumber = 0; frameNumber < imageSequenceNode->GetNumberOfDataNodes(); frameNumber++ ) // Iterate including the first frame
+  {
+    std::string indexValue = masterSequenceNode->GetNthIndexValue( frameNumber );
+
+    // Check the image slice is OK
+    sliceNode = vtkMRMLVolumeNode::SafeDownCast( imageSequenceNode->GetDataNodeAtValue( indexValue.c_str() ) );
+    if ( sliceNode == NULL || sliceNode->GetImageData() == NULL )
+    {
+      return;
+    }
+    sliceImageData = sliceNode->GetImageData();
+    if ( sliceImageData->GetDimensions()[ 0 ] != sliceDimensions[ 0 ] || sliceImageData->GetDimensions()[ 1 ] != sliceDimensions[ 1 ] || sliceImageData->GetDimensions()[ 2 ] != 1 
+      || sliceImageData->GetSpacing()[ 0 ] != sliceSpacing[ 0 ] || sliceImageData->GetSpacing()[ 1 ] != sliceSpacing[ 1 ] || sliceImageData->GetSpacing()[ 2 ] != sliceSpacing[ 2 ] )
+    {
+      return;
+    }
+
+    unsigned char* startPtr=(unsigned char*)imageData->GetScalarPointer(0, 0, frameNumber);
+    memcpy(startPtr, sliceImageData->GetScalarPointer(), sliceSize);
+  }
+
+#ifdef ENABLE_PERFORMANCE_PROFILING
+  vtkSmartPointer<vtkTimerLog> timer=vtkSmartPointer<vtkTimerLog>::New();
+  timer->StartTimer();  
+#endif
+  vtkSmartPointer< vtkMetaImageWriter > imageWriter = vtkSmartPointer< vtkMetaImageWriter >::New();
+  imageWriter->SetInputData( imageData );
+  imageWriter->SetFileName( fileName.c_str() ); // This automatically takes care of the file extensions. Note: Saving to .mhd is way faster than saving to .mha
+  imageWriter->SetCompression( false );
+  imageWriter->Write();
+#ifdef ENABLE_PERFORMANCE_PROFILING
+  timer->StopTimer();
+  vtkWarningMacro("Image writing: " << timer->GetElapsedTime() << "sec\n");
```

Comment:
`vtkInfoMacro` ?     (Now available in [Slicer/Libs/vtkAddon/vtkLoggingMacros.h](https://github.com/Slicer/Slicer/blob/master/Libs/vtkAddon/vtkLoggingMacros.h)